File: OfficeManagementSystem/oms-app-client/src/app/_services/user.service.ts
import { Injectable } from "@angular/core";
import { HttpClient } from "@angular/common/http";
import { User } from "../_models";

@Injectable()
export class UserService{
    constructor(private http: HttpClient){}

    getUsers(){
        return this.http.get<User[]>('/oms/users');
    }

    getUserById(id: number){
        return this.http.get('/oms/users/' + id);
    }

    createUser(user: User){
        return this.http.post('/oms/users', user);
    }

    updateUser(user: User){
        return this.http.put('/oms/users/' + user.id, user);
    }

    deleteUser(id: number){
        return this.http.delete("/oms/delete/" + id);
    }
}

File: OfficeManagementSystem/oms-app-server/src/main/java/com/rk/oms/service/UserService.java
package com.rk.oms.service;

import java.util.List;

import com.rk.oms.model.User;

public interface UserService {
	List<User> getUsers();

	User getUser(Long userId);

	User createUser(User user);

	User updateUser(Long userId, User user);

	int deleteUser(Long userId);
	
	User findUser(User user);

	List<User> findUsers(String userNameStr);

}
